internal/stream/stream.go

```diff
@@ -317,7 +317,20 @@ func NewPushStream(ctx context.Context, nodeID string, subName string, subscript
 	// set http connection timeout from the subscription
 	if subModel.AckDeadlineSeconds != 0 {
 		// make sure to convert sec to milli-sec
-		config.ConnectTimeoutMS = int(subModel.AckDeadlineSeconds) * 1e3
+		// set the timeout value only if greater than the default
+		ackDeadlineMs := int(subModel.AckDeadlineSeconds) * 1000
+		if ackDeadlineMs > config.ConnectTimeoutMS {
+			config.ConnectTimeoutMS = ackDeadlineMs
```

Comment:
Do we intend to set both values here?

Reply:
One is a TCP connection establishment timeout and the other is the HTTP timeout.
We will need both of them. Ideally TCP connection establishment won't take much time but in case if it does, we should be able to handle it as well.

Reply:
TCP connection establishment timeout should not be considered from AckDeadline right? AckDeadline is specific to per message delivery.

Reply:
So this comes into picture when we publish to http endpoint. The default is 2s but since we were updating it with subscription deadline ack secs (which is 10s), not setting this would mean we are reducing it.. Kind of risky